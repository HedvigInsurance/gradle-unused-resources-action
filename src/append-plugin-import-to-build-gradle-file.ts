import * as fs from 'fs-extra'

const APPLY_IMPORT = `
apply(from="unused.gradle.kts")
`

const UNUSED_FILE_CONTENTS = `
buildscript {
  repositories {
      gradlePluginPortal()
  }
  dependencies { classpath("gradle.plugin.com.github.konifar.gradle:plugin:0.3.3") }
}
apply<com.github.konifar.gradle.remover.UnusedResourcesRemoverPlugin>()
`

export const appendPluginImportToBuildGradleFile = async (): Promise<void> => {
    await fs.appendFile('./build.gradle.kts', APPLY_IMPORT)
    await fs.writeFile('./unused.gradle.kts', UNUSED_FILE_CONTENTS)
}
