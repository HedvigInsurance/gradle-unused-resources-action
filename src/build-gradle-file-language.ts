export type BuildGradleFileLanguage = 'groovy' | 'kotlinscript'
export const buildGradleFileLanguage = (
  buildGradlePath: string
): BuildGradleFileLanguage => {
  if (buildGradlePath.endsWith('kts')) {
    return 'kotlinscript'
  }
  if (buildGradlePath.endsWith('gradle')) {
    return 'groovy'
  }

  throw new Error(
    `Unknown build.gradle-file language for file: ${buildGradlePath}`
  )
}
