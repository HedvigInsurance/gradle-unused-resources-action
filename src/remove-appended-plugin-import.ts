import * as fs from 'fs-extra'

export const removeAppendedPluginImport = async (
  oldBuildGradleKtsContent: string
): Promise<void> => {
  await fs.writeFile('./build.gradle.kts', oldBuildGradleKtsContent)
  await fs.remove('./unused.gradle.kts')
}
