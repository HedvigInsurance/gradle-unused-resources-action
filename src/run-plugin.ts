import * as exec from '@actions/exec'

export const runPlugin = async (): Promise<void> => {
    exec.exec('./gradlew removeUnusedResources')
}