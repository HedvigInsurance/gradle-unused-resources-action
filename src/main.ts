import * as core from '@actions/core'
import { appendPluginImportToBuildGradleFile } from './append-plugin-import-to-build-gradle-file'

async function run(): Promise<void> {
  try {
    appendPluginImportToBuildGradleFile()
  } catch (error) {
    if (error instanceof Error) core.setFailed(error.message)
  }
}

run()
