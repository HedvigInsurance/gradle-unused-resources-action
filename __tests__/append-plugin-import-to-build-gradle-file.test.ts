import * as fs from 'fs-extra'
import {vol} from 'memfs'
import {expect, it, beforeEach, jest} from '@jest/globals'
import {appendPluginImportToBuildGradleFile} from '../src/append-plugin-import-to-build-gradle-file'

jest.mock('fs')

const MOCK_BUILD_GRADLE_KTS = `
buildscript {
  repositories {
    mavenCentral()
  }
}
`

beforeEach(() => {
  vol.reset()
  vol.mkdirSync('.', {recursive: true})
})

it('should append required module configuration correctly', async () => {
  await fs.writeFile('./build.gradle.kts', MOCK_BUILD_GRADLE_KTS)

  await appendPluginImportToBuildGradleFile()

  const buildGradleKts = (await fs.readFile('./build.gradle.kts')).toString()
  const unusedGradleKts = (await fs.readFile('./unused.gradle.kts')).toString()

  expect(buildGradleKts).toMatchSnapshot()
  expect(unusedGradleKts).toMatchSnapshot()
})
