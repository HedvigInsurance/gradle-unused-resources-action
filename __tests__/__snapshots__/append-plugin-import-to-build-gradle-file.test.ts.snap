// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should append required module configuration correctly 1`] = `
"
buildscript {
  repositories {
    mavenCentral()
  }
}

apply(from=\\"unused.gradle.kts\\")
"
`;

exports[`should append required module configuration correctly 2`] = `
"
buildscript {
  repositories {
      gradlePluginPortal()
  }
  dependencies { classpath(\\"gradle.plugin.com.github.konifar.gradle:plugin:0.3.3\\") }
}
apply<com.github.konifar.gradle.remover.UnusedResourcesRemoverPlugin>()
"
`;
